src/function/throttle.ts

```diff
@@ -24,7 +24,10 @@
  *   throttledFunction(); // Will log 'Function executed'
  * }, 1000);
  */
-export function throttle<F extends (...args: any[]) => void>(func: F, throttleMs: number): F {
+export function throttle<F extends (...args: Parameters<F>) => ReturnType<F>>(
```

Comment:
```suggestion
export function throttle<F extends (...args: any[]) => void>(
```

I think `F` should extend `(...args: any[]) => void` here; Otherwise it references itself.

Reply:
I was also unsure about whether it was the right decision to make these changes, but thank you for reviewing it!